### DISCUSSION TITLE
Keyboard Navigation not working properly

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#6293] [Q&A]
Hi,

I was testing Keyboard Accessibility by using the `Tab` & `Shift +Tab` keys to navigate through the controls. I have a Datagrid with a checkbox as one of the columns (code pasted below). The order of the navigation is not correct and it keeps on jumping through the rows. I have also attached a small video for reference. Is there a way to ensure that pressing the `Tab` key navigates through the control in the same way as it is seen in the UI?

**Code for Datagrid**
```
<DataGrid Name="peopleDataGrid" Items="{Binding People}" SelectedItem="{Binding SelectedPerson, Mode=TwoWay}" SelectedIndex="{Binding SelectedIndexValue, Mode=OneWayToSource}" DockPanel.Dock="Left">
      <DataGrid.ContextMenu>
        <ContextMenu PlacementMode="Bottom">
          <MenuItem Header="Go to next odd employee no." Command="{Binding GoToNextOddEmployeeNo}" InputGesture="Alt+C">
          </MenuItem>
        </ContextMenu>
       </DataGrid.ContextMenu>
      <DataGrid.KeyBindings>
        <KeyBinding Gesture="Alt+C" Command="{Binding GoToNextOddEmployeeNo}"/>
      </DataGrid.KeyBindings>
      <DataGrid.Columns>
        <DataGridTemplateColumn>
          <DataGridTemplateColumn.CellTemplate>
            <DataTemplate>
                <CheckBox Name="IsChecked" HorizontalAlignment="Center" VerticalAlignment="Center"/>
            </DataTemplate>
          </DataGridTemplateColumn.CellTemplate>
        </DataGridTemplateColumn>
        <DataGridTextColumn Header="Employee Number" Binding="{Binding EmployeeNumber}" Width="1*" />
        <DataGridTextColumn Header="First Name" Binding="{Binding FirstName}" Width="2*" />
        <DataGridTextColumn Header="Last Name" Binding="{Binding LastName}" Width="2*" />
        <DataGridTextColumn Header="Department" Binding="{Binding DepartmentNumber}" Width="*" />
      </DataGrid.Columns>
    </DataGrid>
```

![KeyboardAccessibility](https://user-images.githubusercontent.com/79670853/126749192-a7f987f0-c6ba-48ee-8750-92dfed85a369.gif)

**Comment by ShrutiJaiswal1494** [comment]
Can someone please help me with this.?

**Comment by hailstorm75** [comment]
The issue is probably caused by your duplicate entries. You have to either make sure that there are no duplicate instances in your `People` collection, or you'll have to modify the type contained in the `People` collection - add a unique identifier, override the `Equals` method...
